rts/Sim/Units/UnitTypes/Builder.cpp

```diff
@@ -722,16 +722,15 @@ bool CBuilder::StartBuild(BuildInfo& buildInfo, CFeature*& feature, bool& inWait
 			// closest to us
 			const CGroundBlockingObjectMap::BlockingMapCell& cell = groundBlockingObjectMap.GetCellUnsafeConst(buildInfo.pos);
 
-			const CSolidObject* o = nullptr;
 			const CUnit* u = nullptr;
 
 			// look for any blocking assistable buildee at build.pos
 			for (size_t i = 0, n = cell.size(); i < n; i++) {
-				const CUnit* cu = dynamic_cast<const CUnit*>(o = cell[i]);
+				const CUnit* cu = dynamic_cast<const CUnit*>(cell[i]);
```

Comment:
why even const cast?

Reply:
`const` has to be in the cast because `cu` is `const *`, and `cu` being `const *` is probably a decent idea because we don't want to modify it. It's type repetition though, how about
```suggestion
				auto cu = dynamic_cast<const CUnit*>(cell[i]);
```